rust/main/chains/hyperlane-ton/src/types/message.rs

```diff
@@ -0,0 +1,59 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+#[derive(Debug, Deserialize)]
+pub struct MessageResponse {
+    pub address_book: Option<HashMap<String, Address>>,
+    pub messages: Vec<Message>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Address {
+    #[serde(rename = "user_friendly")]
+    pub user_friendly: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Message {
+    pub bounce: Option<bool>,
+    pub bounced: Option<bool>,
+    pub created_at: String,
+    pub created_lt: String,
+    pub destination: Option<String>,
+    pub fwd_fee: Option<String>,
+    pub hash: String,
+    pub ihr_disabled: Option<bool>,
+    pub ihr_fee: Option<String>,
+    pub import_fee: Option<String>,
+    pub init_state: Option<MessageState>,
+    pub message_content: MessageContent,
+    pub opcode: Option<String>,
+    pub source: Option<String>,
+    pub value: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct MessageState {
+    pub body: String,
+    pub decoded: Option<DecodedMessage>,
+    pub hash: String,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct MessageContent {
+    pub body: String,
+    pub decoded: Option<DecodedMessage>,
+    pub hash: String,
+}
```

Comment:
I wonder what the difference is between these two struct?

Reply:
`MessageState` is removed